core/jvmMain/src/kotlinx/serialization/internal/Platform.kt

```diff
@@ -51,11 +51,19 @@ internal actual fun <T : Any> KClass<T>.constructSerializerForGivenTypeArgs(vara
         null
     }
     if (fromNamedCompanion != null) return fromNamedCompanion
+    //Check globally registered serializers
+    jvmGlobalSerializer(jClass)?.let { return it }
     // Check for polymorphic
     return polymorphicSerializer()
 }
 
-private fun <T: Any> Class<T>.isNotAnnotated(): Boolean {
+private fun <T> jvmGlobalSerializer(jClass: Class<T>): KSerializer<T>? =
+        when (jClass) {
+            UUID::class.java -> UUIDSerializer as? KSerializer<T>
+            else -> null
+        }
```

Comment:
This is not a scalable solution, more significantly it doesn't work in a way that keeps parity between standard and custom serializers. In addition, this code path is only for reflectively resolving serializers. In that case the serialmodule is available. The method is mainly for the code path with type arguments, which UUID doesn't have. As such it will result in surprising and inconsistent behaviour: `serializer(typeOf<UUID>())` will not return the serializer because it has no type arguments. The place that you would add the additional built-in would be in `Primitives.kt` line 69, `builtinSerializerOrNull`. But frankly the approach I would take would be to create a different method that uses serializerOrNull and then adds whatever you want to add as alternative. Maybe even better is to just use a `ContextSerializer` and then you create a module which adds all these defaults.

Of course kotlinx.serialization is primarily focused on statically (at compile time) resolving serializers and for that you need to announce the serializers to the compiler with the right annotations. It may be worthwhile to allow libraries to declare serializers in a way that they are automatically picked up by the compiler but that is a whole different thing.

Reply:
Actually, it will work with `serializer<UUID>()` (as it shown in tests), because this method resolves all serializers. 

However, the bigger problem that this method used only for top-level reflective resolving (serializer<> function) and it does not affect static compile-time resolving. That is, any class that contains UUID (e.g. `@Serializable data class MyData(val uuid: UUID)`) will report compile-time exception that no serializer is found, require using `@UseSerializers/SerializersModule`, so original issue is kinda not solved.